### PR TITLE
Feature/set syncable

### PR DESCRIPTION
**Pull request by kalikaneko**
set_syncable method + service tokens

### CODE REVIEW

**Comment by kalikaneko** [comment]
WIP, this is needed to test latest core/cli

File: client/src/leap/soledad/client/api.py [hunk 1]
```diff
@@ -911,6 +931,35 @@ def raw_sqlcipher_operation(self, *args, **kw):
...
+
+    @inlineCallbacks
+    def get_or_create_service_token(self, service):
+        """
+        Return the stored token for a given service, or generates and stores a
+        random one if it does not exist.
+
+        These tokens can be used to authenticate services.
+        """
+        yield self.create_index('by-servicetoken', 'type', 'service')
```
Comment (meskio) [inline client/src/leap/soledad/client/api.py]: Do we need to sync the tokens between devices? I have the feeling that the token should be per device, but I'm not sure if I have good reasons for that.

File: client/src/leap/soledad/client/api.py [hunk 2]
```diff
@@ -209,12 +211,15 @@ def __init__(self, uuid, passphrase, secrets_path, local_db_path,
         self._crypto = SoledadCrypto(self._secrets.remote_storage_secret)
         self._init_u1db_sqlcipher_backend()
 
+        self.syncable = syncable
```
Comment (meskio) [inline client/src/leap/soledad/client/api.py]: I find confusing that `syncable` exist also in `shared_db` do we need it in both places?

File: client/src/leap/soledad/client/api.py [hunk 3]
```diff
@@ -652,6 +657,21 @@ def uuid(self):
...
+        This can be used to start a database with offline state and switch it
+        online afterwards. Or the opposite: stop syncs when connection is lost.
+
+        :param syncable: new status for syncable.
+        :type syncable: bool
+        """
+        # TODO should check that we've got a token!
+        self.shared_db.syncable = syncable
+        if syncable and not self._dbsyncer:
+            self._init_u1db_syncer()
```
Comment (meskio) [inline client/src/leap/soledad/client/api.py]: If we keep `self.syncable` maybe we should set it here.

File: client/src/leap/soledad/client/api.py (hunk 2 above)
Comment (meskio) [inline client/src/leap/soledad/client/api.py]: If is needed maybe we could do:
```
@property
def syncable(self):
      return self.shared_db.syncabe
```

Comment (kalikaneko) [inline client/src/leap/soledad/client/api.py]: fixing, I think we don't need it as an attribute in Soledad

File: client/src/leap/soledad/client/api.py (hunk 1 above)
Comment (kalikaneko) [inline client/src/leap/soledad/client/api.py]: agree, it reduces the chance of creating and syncing different tokens.
left as a TODO note, i'd like to cover smtp auth asap and move on.

**Comment by kalikaneko** [comment]
squashed, and signed

**Pull request merged**